src/Sparrow/ObjectPool.cs

```diff
@@ -103,7 +103,75 @@ private struct CacheAwareElement
             private T Value4;
 
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            public bool TryClaim(ref CacheAwareElement bucket, out T item)
+            public static void Clear<TEvictionStrategy>(ref CacheAwareElement bucket, ref TEvictionStrategy policy)
+                where TEvictionStrategy : struct, IEvictionStrategy<T>
+            {
+                // Note that the initial read is optimistically not synchronized. That is intentional. 
+                // We will interlock only when we have a candidate. in a worst case we may miss some
+                // recently returned objects. Not a big deal.
```

Comment:
That means you can miss it and either not finding there is an object there to reclaim or in the case of allocate create a new one even if a thread has a nanosecond before returned one. Very very very unlikely to happen but you still need the guard to avoid losing objects for disposition. That means that after the clear you have 1 object in the pool instead of 0